Update cargo

weihanglo opened the pull request:
8 commits in 2381cbdb4e9b07090f552d34a44a529b6e620e44..8c460b2237a6359a7e3335890db8da049bdd62fc
2022-12-23 12:19:27 +0000 to 2023-01-04 14:30:01 +0000
- test: revive nightly plugin tests to work (rust-lang/cargo#11534)
- Add note to release notes about rejecting multiple registries. (rust-lang/cargo#11531)
- Fix a typo `fresheness` -&gt; `freshness` (rust-lang/cargo#11529)
- Reasons for rebuilding (rust-lang/cargo#11407)
- Asymmetric tokens (rust-lang/cargo#10771)
- Use proper git URL for GitHub repos (rust-lang/cargo#11517)
- Add `registry.default` example (rust-lang/cargo#11516)
- Support vendoring with different revs from same git repo (rust-lang/cargo#10690)

Also update license exceptions and permitted dependencies
for new cargo dependency "pasetors".

A new dependency `getrandom` is added into `rustc-workspace-hacks`,
since it requires feature `js`.


r? @ghost

weihanglo commented:
@ehuss, please help review this update. I am not quite sure the change of `PERMITTED_RUSTC_DEPENDENCIES`. Looks like `pasetors` enables `js` feature by default.

ehuss commented:
Hm, that's unfortunate.

I'm wondering if it would be a good idea to filter out wasm32 from the permitted list? WDYT? I hacked up a solution here: https://github.com/ehuss/rust/commit/a9ec95bdf7705e206ea46425bc2c012e566b85b7

That commit also fixes a bug where it was not removing some extra entries from the permitted list.

I'd like to rope in @Mark-Simulacrum to see if you think it would be reasonable to filter out wasm32 from the tidy permitted dependency list for rustc and cranelift?  The idea is to keep the list cleaner by removing dependencies that would never actually be used in rustc.

weihanglo commented:
That looks neat to me!

If we need more discussions on a proper fix, should we not block this PR and proceed?

ehuss commented:
That's fair. I'll go ahead and approve this, and then follow up with a PR with my proposed change to keep the tidy list tidier.

@bors r+ p=1 rollup=never

bors commented:
:pushpin: Commit 323938c34f99801bcd3577e8c8a5816c3b73f9f7 has been approved by `ehuss`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=ehuss 323938c34f99801bcd3577e8c8a5816c3b73f9f7 -->
<!-- homu: {"type":"Approved","sha":"323938c34f99801bcd3577e8c8a5816c3b73f9f7","approver":"ehuss","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

ehuss commented:
Looks like some issue with bors, it didn't actually get into the queue.

bors commented:
:hourglass: Testing commit 323938c34f99801bcd3577e8c8a5816c3b73f9f7 with merge 6bf7396ed651febcd9e89831eebe23f8e560142d...
<!-- homu: {"type":"BuildStarted","head_sha":"323938c34f99801bcd3577e8c8a5816c3b73f9f7","merge_sha":"6bf7396ed651febcd9e89831eebe23f8e560142d"} -->

rust-log-analyzer commented:
The job **`dist-mips64el-linux`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/3832723635/jobs/6523368065) [(plain)](https://github.com/rust-lang-ci/rust/commit/6bf7396ed651febcd9e89831eebe23f8e560142d/checks/10421320623/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] build_script_build test:false 0.226
   Compiling openssl v0.10.38
   Compiling os_str_bytes v6.0.0
[RUSTC-TIMING] build_script_build test:false 0.333
   Compiling pem-rfc7468 v0.6.0
[RUSTC-TIMING] build_script_build test:false 0.376
[RUSTC-TIMING] build_script_build test:false 0.386
[RUSTC-TIMING] build_script_build test:false 0.219
[RUSTC-TIMING] const_oid test:false 0.471
---
[RUSTC-TIMING] proc_macro2 test:false 3.644
[RUSTC-TIMING] proc_macro_error_attr test:false 0.743
   Compiling ppv-lite86 v0.2.8
[RUSTC-TIMING] build_script_build test:false 0.390
[RUSTC-TIMING] pem_rfc7468 test:false 1.208
[RUSTC-TIMING] crc32fast test:false 0.219
[RUSTC-TIMING] crc32fast test:false 0.219
   Compiling spki v0.6.0
   Compiling strsim v0.10.0
[RUSTC-TIMING] hashbrown test:false 0.804
   Compiling pkcs8 v0.9.0
[RUSTC-TIMING] heck test:false 0.327
---
   Compiling openssl-src v111.22.0+1.1.1q
[RUSTC-TIMING] same_file test:false 0.194
   Compiling hmac v0.12.1
[RUSTC-TIMING] build_script_build test:false 0.332
   Compiling hkdf v0.12.3
[RUSTC-TIMING] hmac test:false 0.123
[RUSTC-TIMING] strsim test:false 1.498
   Compiling sha2 v0.10.6
[RUSTC-TIMING] openssl_src test:false 0.283
   Compiling walkdir v2.3.2
   Compiling walkdir v2.3.2
[RUSTC-TIMING] signature test:false 0.110
   Compiling tar v0.4.38
[RUSTC-TIMING] hkdf test:false 0.115
   Compiling rand_chacha v0.3.0
[RUSTC-TIMING] spki test:false 1.642
   Compiling bitmaps v2.1.0
[RUSTC-TIMING] rand_chacha test:false 0.331
   Compiling rfc6979 v0.3.1
[RUSTC-TIMING] filetime test:false 0.301
[RUSTC-TIMING] filetime test:false 0.301
   Compiling elliptic-curve v0.12.3
[RUSTC-TIMING] pkcs8 test:false 1.816
[RUSTC-TIMING] rfc6979 test:false 0.083
   Compiling clap_lex v0.2.2
[RUSTC-TIMING] cc test:false 1.544
   Compiling libz-sys v1.1.3
   Compiling libz-sys v1.1.3
[RUSTC-TIMING] socket2 test:false 1.219
   Compiling openssl-sys v0.9.72
[RUSTC-TIMING] elliptic_curve test:false 0.348
[RUSTC-TIMING] build_script_build test:false 0.417
   Compiling curl-sys v0.4.59+curl-7.86.0
[RUSTC-TIMING] build_script_build test:false 0.398
[RUSTC-TIMING] sec1 test:false 1.789
---
   Compiling static_assertions v1.1.0
[RUSTC-TIMING] build_script_build test:false 0.557
   Compiling time-core v0.1.0
[RUSTC-TIMING] build_script_build test:false 0.266
   Compiling fiat-crypto v0.1.17
[RUSTC-TIMING] either test:false 0.131
   Compiling textwrap v0.15.0
[RUSTC-TIMING] proc_macro_error test:false 0.469
   Compiling vte v0.3.3
---
[RUSTC-TIMING] time test:false 7.534
[RUSTC-TIMING] tar test:false 2.580
[RUSTC-TIMING] combine test:false 7.657
[RUSTC-TIMING] serde_derive test:false 6.035
   Compiling orion v0.17.3
[RUSTC-TIMING] fiat_crypto test:false 8.270
[RUSTC-TIMING] serde_derive test:false 6.324
   Compiling serde v1.0.152
[RUSTC-TIMING] libz_sys test:false 0.067
   Compiling flate2 v1.0.23
---
[RUSTC-TIMING] rustfix test:false 1.253
[RUSTC-TIMING] serde_json test:false 3.529
[RUSTC-TIMING] semver test:false 0.832
[RUSTC-TIMING] os_info test:false 1.053
[RUSTC-TIMING] pasetors test:false 2.729
[RUSTC-TIMING] libssh2_sys test:false 0.134
[RUSTC-TIMING] curl_sys test:false 0.129
   Compiling crates-io v0.35.1 (/checkout/src/tools/cargo/crates/crates-io)
   Compiling crypto-hash v0.3.4
---
[RUSTC-TIMING] rustc_workspace_hack test:false 0.037
   Compiling rls v2.0.0 (/checkout/src/tools/rls)
[RUSTC-TIMING] rls test:false 0.898
    Finished release [optimized] target(s) in 32.46s
duplicate artifacts found when compiling a tool, this typically means that something was recompiled because a transitive dependency has different features activated than in a previous build:
the following dependencies have different features:
the following dependencies have different features:
  getrandom 0.2.8 (registry+https://github.com/rust-lang/crates.io-index)
    `rls` additionally enabled features {} at "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/mips64el-unknown-linux-gnuabi64/release/deps/libgetrandom-fe41900f8ca686fb.rlib"
    `cargo` additionally enabled features {"wasm-bindgen", "js", "js-sys"} at "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/mips64el-unknown-linux-gnuabi64/release/deps/libgetrandom-704177f6a79dc3e8.rlib"


to fix this you will probably want to edit the local src/tools/rustc-workspace-hack/Cargo.toml crate, as that will update the dependency graph to ensure that these crates all share the same feature set
thread 'main' panicked at 'tools should not compile multiple copies of the same crate', tool.rs:201:13
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/3832723635/jobs/6523644583)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/3832723635/jobs/6523644583","builder_name":"checks-actions"} -->

weihanglo commented:
Ohh no. Now we need to force everybody to enable features `["wasm-bindgen", "js", "js-sys"]` for `getrandom`. Does this still look good?

klensy commented:
> Ohh no. Now we need to force everybody to enable features `["wasm-bindgen", "js", "js-sys"]` for `getrandom`. Does this still look good?

Not nice. Why not disable this features in newly introduced dependency (or pr/ask maintainer to do that)?

ehuss commented:
Enabling `js` on `getrandom` in the workspace-hack seems fine. Those dependencies aren't otherwise built, and it will have no effect.

weihanglo committed:
Update cargo

8 commits in 2381cbdb4e9b07090f552d34a44a529b6e620e44..8c460b2237a6359a7e3335890db8da049bdd62fc
2022-12-23 12:19:27 +0000 to 2023-01-04 14:30:01 +0000
- test: revive nightly plugin tests to work (rust-lang/cargo#11534)
- Add note to release notes about rejecting multiple registries. (rust-lang/cargo#11531)
- Fix a typo `fresheness` -&gt; `freshness` (rust-lang/cargo#11529)
- Reasons for rebuilding (rust-lang/cargo#11407)
- Asymmetric tokens (rust-lang/cargo#10771)
- Use proper git URL for GitHub repos (rust-lang/cargo#11517)
- Add `registry.default` example (rust-lang/cargo#11516)
- Support vendoring with different revs from same git repo (rust-lang/cargo#10690)

Also update license exceptions and permitted dependencies
for new cargo dependency "pasetors".

A new dependency `getrandom` is added into `rustc-workspace-hacks`,
since it requires feature `js`.

weihanglo commented:
Enabled that with a rebase. Will merge it now.
Feel free to cancel if I did it wrong.

@bors r+ p=1 rollup=never

bors commented:
:pushpin: Commit 6306fd3be407fa526785d2222f4cb00ed1f01abf has been approved by `weihanglo`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=weihanglo 6306fd3be407fa526785d2222f4cb00ed1f01abf -->
<!-- homu: {"type":"Approved","sha":"6306fd3be407fa526785d2222f4cb00ed1f01abf","approver":"weihanglo","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 6306fd3be407fa526785d2222f4cb00ed1f01abf with merge 659e169d37990b9c730a59a96081f2ef7afbe8f1...
<!-- homu: {"type":"BuildStarted","head_sha":"6306fd3be407fa526785d2222f4cb00ed1f01abf","merge_sha":"659e169d37990b9c730a59a96081f2ef7afbe8f1"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/3840702554/jobs/6542405022)
Approved by: weihanglo
Pushing 659e169d37990b9c730a59a96081f2ef7afbe8f1 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"weihanglo","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/3840702554/jobs/6542405022"},"merge_sha":"659e169d37990b9c730a59a96081f2ef7afbe8f1"} -->

rust-timer commented:
Finished benchmarking commit (659e169d37990b9c730a59a96081f2ef7afbe8f1): [comparison URL](https://perf.rust-lang.org/compare.html?start=b7cdb635c4b973572307ad288466fba64533369c&end=659e169d37990b9c730a59a96081f2ef7afbe8f1&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=b7cdb635c4b973572307ad288466fba64533369c&end=659e169d37990b9c730a59a96081f2ef7afbe8f1&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=b7cdb635c4b973572307ad288466fba64533369c&end=659e169d37990b9c730a59a96081f2ef7afbe8f1&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -3.2% | [-3.2%, -3.2%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=b7cdb635c4b973572307ad288466fba64533369c&end=659e169d37990b9c730a59a96081f2ef7afbe8f1&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.
